Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -1342,39 +1339,296 @@ namespace OpenInfraPlatform
+					std::vector<carve::geom::vector<3>>& segmentStartPoints,
+					const std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcTrimmingSelect>>& trim1Vec,
+					const std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcTrimmingSelect>>& trim2Vec,
+					const bool senseAgreement,
+					const typename IfcEntityTypesT::IfcTrimmingPreference& trimmingPreference
+				) const noexcept(false)
+				{
+					// **************************************************************************************************************************
+					//	https://standards.buildingsmart.org/IFC/RELEASE/IFC4_3/lexical/IfcOffsetCurve2D.htm
+					//	ENTITY IfcOffsetCurve2D
+					//		SUBTYPE OF(IfcOffsetCurve);
+					//			Distance: IfcLengthMeasure;
+					//			SelfIntersect: IfcLogical;
+					//		WHERE
+					//			DimIs2D : BasisCurve.Dim = 2;
+					//	END_ENTITY;
+					// **************************************************************************************************************************
+					EXPRESSReference<typename IfcEntityTypesT::IfcCurve> basisCurve = offsetCurve2D->BasisCurve;
+					std::vector<carve::geom::vector<3>> basisCurvePoints;
+					double distance = offsetCurve2D->Distance * this->UnitConvert()->getLengthInMeterFactor();
+
+					convertIfcCurve(basisCurve, basisCurvePoints, segmentStartPoints,
+						trim1Vec, trim2Vec, senseAgreement, trimmingPreference);
+
+					std::vector<carve::geom::vector<3>> offsetCurve2DPoints;
+					carve::geom::vector<3> basisCurveDirection = basisCurvePoints[basisCurvePoints.size() - 1] - basisCurvePoints[0];
+
+					for (int i = 0; i < basisCurvePoints.size() - 1; i++) {
+						carve::geom::vector<3> startPoint = basisCurvePoints[i];
+						carve::geom::vector<3> endPoint = basisCurvePoints[i + 1];
+						carve::geom::vector<3> offsetPoint;
+
+						if (this->GeomSettings()->areEqual(basisCurveDirection.z, 0.0)) {
+							carve::geom::vector<2> startPoint2D = carve::geom::VECTOR(startPoint.x, startPoint.y);
+							carve::geom::vector<2> endPoint2D = carve::geom::VECTOR(endPoint.x, endPoint.y);
+
+							carve::geom::vector<2> tangent = endPoint2D - startPoint2D;
+							carve::geom::vector<2> orthogonal = carve::geom::VECTOR(-tangent.y, tangent.x);
+
+							offsetPoint = carve::geom::VECTOR(startPoint.x + distance * orthogonal.x,
+								startPoint.y + distance * orthogonal.y,
+								startPoint.z);
+						}
+						else if (this->GeomSettings()->areEqual(basisCurveDirection.y, 0.0)) {
+							carve::geom::vector<2> startPoint2D = carve::geom::VECTOR(startPoint.x, startPoint.z);
+							carve::geom::vector<2> endPoint2D = carve::geom::VECTOR(endPoint.x, endPoint.z);
+
+							carve::geom::vector<2> tangent = endPoint2D - startPoint2D;
+							carve::geom::vector<2> orthogonal = carve::geom::VECTOR(-tangent.y, tangent.x);
+
+							offsetPoint = carve::geom::VECTOR(startPoint.x + distance * orthogonal.x,
+								startPoint.y,
+								startPoint.z + distance * orthogonal.y);

```

Comment:
Maybe needs to be changed

Reply:
Possible solution: [e4565ca](https://github.com/tumcms/Open-Infra-Platform/pull/558/commits/e4565ca74f0e961fa1a01a283695f6f60e536c07)